src/xray/askui-xray-step-reporter.ts

```diff
@@ -180,20 +182,27 @@ export class AskUIXRayStepReporter implements Reporter {
       if (!(fs.existsSync(this.outputDirectory))) {
         fs.mkdirSync(this.outputDirectory, { recursive: true });
       }
-
-      let existingData: { tests: XRayTestObject[]} = { tests: []};
-      if (fs.existsSync(outputFilePath)) {
-        const fileContent = fs.readFileSync(outputFilePath, { encoding: 'utf-8' });
-        existingData = JSON.parse(fileContent);
+      if (!(fs.existsSync(outputFilePath))) {
+        fs.writeFileSync(outputFilePath, '{"tests":[]}');
       }
-      existingData.tests.push(...this.result);
-
-      fs.writeFileSync(
-        outputFilePath,
-        JSON.stringify(existingData, null, 2));
+      lock(outputFilePath)
+        .then((release) => {
+          let existingData: { tests: XRayTestObject[]} = { tests: []};
+          if (fs.existsSync(outputFilePath)) {
+            const fileContent = fs.readFileSync(outputFilePath, { encoding: 'utf-8' });
+            existingData = JSON.parse(fileContent);
+          }
+          existingData.tests.push(...this.result);
+    
+          fs.writeFileSync(
+            outputFilePath,
+            JSON.stringify(existingData, null, 2));
+          
+          return release();
+        });
```

Comment:
I would implement it in this way. So the Exception handling it done in a proper way. Otherwise the lockfile get never released.

I have still my thoughts. What's happend, when the Process crashes and the lockfile does not get released. How can we clean this up? Manually Can we move this condition during constructor? So the user get's notified before he executes the test and not afterwards?

```suggestion
      let release;
      try{
        release = await lock(outputFilePath)
      }
      catch {
         throw LockFileExecption("Can't lock the output file!")
      }      
      try{
          let existingData: { tests: XRayTestObject[]} = { tests: []};
          if (fs.existsSync(outputFilePath)) {
            const fileContent = fs.readFileSync(outputFilePath, { encoding: 'utf-8' });
            existingData = JSON.parse(fileContent);
          }
          existingData.tests.push(...this.result);
    
          fs.writeFileSync(
            outputFilePath,
            JSON.stringify(existingData, null, 2));
          
          }
          catch{
            throw ReportReadWriteException("blbalblab")
            
          }
          finally{                      
             release && release();
          }
```

Reply:
We can check in the constructor if a stale lockfile exists. There is even a method for that `.check(file, [options])`. But it should just lock it anyway when the lockfile is stale according to the README of `proper-lockfile`.

I will also add a retry so concurrent locks will not lead to an exception right away. 5 seconds should be enough for any for any lock to be released: `await lockfile.lock(`${tmpDir}/foo`, { retries: { retries: 5, maxTimeout: 1000 } });`